### ISSUE TITLE
`no-deprecated` misses deprecated functions in TS lib.dom.d.ts

### ISSUE THREAD

**Issue by JasonKaz** [oxc-project/tsgolint#935]
Disclosure: Generated by AI, but manually tested.

Using `oxlint@1.62.0`, `oxlint-tsgolint@0.22.1`, and `typescript@5.7.2`.

```ts
/**
 * Reproduction: tsgolint `no-deprecated` misses @deprecated from TypeScript's built-in lib.dom.d.ts
 *
 * Run: npx oxlint --type-aware .tmp-repro/no-deprecated-repro.ts
 *
 * Expected: 2 errors (cases A and B)
 * Actual:   1 error  (only case B — user-defined @deprecated is caught)
 */

// ─── CASE A: @deprecated from lib.dom.d.ts — MISSED by tsgolint ──────────────
// navigator.platform is marked @deprecated in TypeScript's built-in lib.dom.d.ts
// @typescript-eslint/no-deprecated: ❌ NOT flagged (expected: error)
const _platform = navigator.platform;

// ─── CASE B: user-defined @deprecated JSDoc — CORRECTLY flagged ──────────────
// @typescript-eslint/no-deprecated: ✅ flagged by tsgolint
/** @deprecated use newFn instead */
function oldFn(): void {}

function newFn(): void {}

oldFn();
newFn();

export {};

```

**Comment by camc314** [comment]
Hi please provide more detail.

What version of typescript are you using?

This property is currently not marked as deprecated.

It was marked as deprecated, but the `@deprecated` tag was removed here; https://github.com/microsoft/TypeScript/pull/62496

**Comment by JasonKaz** [comment]
We are using TypeScript v`5.7.2`.

**Comment by camc314** [comment]
ok, so as I mentioned, the TypeScript team have changed this behaviour in this commit: https://github.com/microsoft/TypeScript/commit/9c4c37704452997d1198ce98c3025d75745d6213

This removes the `@deprecated` tag.

If you do not expect this or disagree with this change, I would reach out directly to them.

For reference, we are using the `v7` typescript libs (most recent). So coming from an older to a newer version might mean changes.

Thanks

**Comment by JasonKaz** [comment]
Shouldn't the linter reference the version that is installed into the application, rather than the version installed with tsgolint? I understand that there is a big version difference in my current situation but there could be differences between patch versions that would impact how this rule functions.

**Comment by camc314** [comment]
see #481 for more information about versioning.

you can use `libReplacement` if you do not wish to use the built in libraries.
